hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -717,7 +718,7 @@ public static HoodieRecord createRecordIndexDelete(String recordKey, String part
     HoodieKey key = new HoodieKey(recordKey, MetadataPartitionType.RECORD_INDEX.getPartitionPath());
     return new HoodieAvroRecord<>(key, isPartitionedRLI
         ? new EmptyHoodieRecordPayloadWithPartition(partitionPath)
-        : new EmptyHoodieRecordPayload());
```

Comment:
we also need to fix createRecordIndexUpdate method at L635 to set `isDeleted` to false for update record.

Reply:
let's add some simple UTs to cover and detect the behavior changes.